### DISCUSSION TITLE
Transparent background for the whole thing?

### DISCUSSION THREAD

**Discussion by sdfg2** [piroor/treestyletab#3220] [Q&A]
How do I make the entire background of the tabs transparent?  I cannot find it after hunting through the DOM.  I have set the background of the sidebar to transparent in FF, the bookmarks and history have a transparent background, but no matter what I do, TST will not set a transparent background.  I even went extreme and tried `body { opacity: 0; }` which removed everything _except_ the background.  Is there something outside of the DOM between FF and TST that we can't touch?

**Comment by irvinm** [comment]
Does this TST CSS work for you?

```
 tab-item tab-item-substance {
   background: transparent !important;
 }
```

**Reply by sdfg2** [reply]
Nope.  I don't think I explained correctly.

![image](https://user-images.githubusercontent.com/5736478/194020533-1bbdb7b9-2a61-42c9-9dc8-9e4bcf0f6f86.png)

That's setting `body { background-color: red; }` so you can see the part I'm talking about.  When I set it to `transparent`, it shows a white background.  I have the sidebar set to be transparent, so I know it's not the sidebar background that's doing it:

![image](https://user-images.githubusercontent.com/5736478/194021314-56fe6b1b-d56e-44f2-a2bd-4f35854584a2.png)

It's almost like there's a shim between the sidebar and TST.  I've gone through the FF DOM for the sidebar and set every parent to `background: transparent !important` and it still shows the white background.

**Reply by irvinm** [reply]
Ah ... I see what you are saying.  So, to be clear ... you are auto-expanding the sidebar?  (Right?)  (Doing something like this:  https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#auto-showhide-sidebar-by-mouseover-hover)

If so, what you are seeing might make sense.  That final section you are seeing is probably the only part that doesn't have any part of the main website page under it.  In doing a little research on this, if you get all the way to the back of the sidebar, there is only white area which is probably what you are seeing.  [In one of the links below, someone talks about how things like history and bookmarks are treated a little differently.]

Of course, if you are not "auto-expanding" ... then I am not being very helpful here. :)

https://www.reddit.com/r/FirefoxCSS/comments/p82ymm/transparent_addon_sidebar/
https://www.reddit.com/r/FirefoxCSS/comments/hrq67y/turn_sidebar_transparent/

**Reply by irvinm** [reply]
You may want to read this over as well:  https://github.com/piroor/treestyletab/issues/2759#issuecomment-727678522

@piroor I assume this really hasn't changed, right?

**Reply by sdfg2** [reply]
No you are quite correct, I am autoexpanding, and I'm very sorry - I swear I did search, I have no idea how I could have missed such an obvious issue title!

It is very disappointing though.
